src/org/labkey/test/tests/DataClassTest.java

```diff
@@ -242,6 +243,67 @@ public void testLongFieldNames()
         assertEquals("Data class grid should have zero rows", 0, sourceHelper.goToDataClass(name).getDataCount());
     }
 
+    @Test
+    public void testFieldUniqueConstraint()
+    {
+        goToProjectHome();
+
+        String dataClassName = "Unique Constraint Test";
+        CreateDataClassPage createPage = goToCreateNewDataClass();
+        createPage.setName(dataClassName);
+
+        log("Add a field with a unique constraint");
+        String fieldName1 = "field Name1";
+        DomainFormPanel domainFormPanel = createPage.getDomainEditor();
+        domainFormPanel.manuallyDefineFields(fieldName1)
+                .setType(FieldDefinition.ColumnType.Integer)
+                .expand().clickAdvancedSettings().setUniqueConstraint(true).apply();
+        log("Add another field with a unique constraint");
+        String fieldName2 = "fieldName_2";
+        domainFormPanel.addField(fieldName2)
+                .setType(FieldDefinition.ColumnType.DateAndTime)
+                .expand().clickAdvancedSettings().setUniqueConstraint(true).apply();
+        log("Add another field which does not have a unique constraint");
+        String fieldName3 = "FieldName@3";
+        domainFormPanel.addField(fieldName3)
+                .setType(FieldDefinition.ColumnType.Boolean);
+        createPage.clickSave();
+
+        viewRawTableMetadata(dataClassName);
+        verifyTableIndices("unique_constraint_test_", List.of("field_name1", "fieldname_2"));
+
+        log("Remove a field unique constraint and add a new one");
+        goToProjectHome();
+        CreateDataClassPage updatePage = goToDataClass(dataClassName);
+        domainFormPanel = updatePage.getDomainEditor();
+        domainFormPanel.getField(fieldName2)
+                .expand().clickAdvancedSettings().setUniqueConstraint(false)
+                .apply();
+        domainFormPanel.getField(fieldName3)
+                .expand().clickAdvancedSettings().setUniqueConstraint(true)
+                .apply();
+        updatePage.clickSave();
+        viewRawTableMetadata(dataClassName);
+        verifyTableIndices("unique_constraint_test_", List.of("field_name1", "fieldname_3"));
+        assertTextNotPresent("unique_constraint_test_fieldname_2");
+    }
+
+    private void viewRawTableMetadata(String dataClassName)
+    {
+        beginAt("/" + EscapeUtil.encode(getProjectName()) + "/query-rawTableMetaData.view?schemaName=exp.data&query.queryName=" + dataClassName);
```

Comment:
`URLBuilder` isn't necessary for non-app urls (no resource path). `WebTestHelper.buildURL` is sufficient for most LKS URLs. It will handle any necessary encoding for you.
```
 beginAt(WebTestHelper.buildURL("query", getProjectName(), "rawTableMetaData", Map.of("schemaName", "exp.data", "query.queryName", dataClassName)));
```

Reply:
Thanks. Done.